src/components/CampaignDetails.vue

```diff
@@ -7,18 +8,45 @@
     <h3> Minimum Contribution </h3>
     <h1>{{ campaign.minContribution }}</h1>
     <p>There is some information missing and this has no styles </p>
+    <input v-model='contribution' placeholder='Enter your contribution' type="number"/>
+    <button :disabled="!contribution"
+    @click="submitContribution">
+    Submit Contribution </button>
+
   </section>
 </template>
 
 <script>
 
+import web3 from "../contracts/web3";
+import ethCampaign from "../contracts/campaign";
+
+web3.eth.getAccounts()
+  .then((accounts) => console.log(accounts[0]))
+
 export default {
   name: 'CampaignDetails',
   props: ['address'],
   computed: {
     campaign() {
       return this.$store.getters.getSingleCampaign(Number(this.address))
     }
+  },
+  created() {
+    const campaignAddress = "0xe8a0980C2B37C2C4FCE45e579301B00CC824bCFc"
+    this.$store.dispatch('fetchCampaign', campaignAddress)
+  },
+  methods: {
+    async submitContribution() {
+      const campaignAddress = "0xe8a0980C2B37C2C4FCE45e579301B00CC824bCFc"
+      const campaignInstance = await ethCampaign.at(campaignAddress);
+      campaignInstance.contribute({ from: this.$store.state.accountNum, value: this.contribution })
+    }
+  },
```

Comment:
Is this what you mean by dynamic campaign addresses?

Reply:
Yeah, I believe this address will be different for each campaign.